lib/src/main/java/xyz/gianlu/librespot/audio/cdn/CdnManager.java

```diff
@@ -178,6 +182,18 @@ void setUrl(@NotNull HttpUrl url) {
                     }
 
                     expiration = expireAt * 1000;
+                    
+                } else if ( expiresStr != null && !expiresStr.isEmpty()) {
```

Comment:
You can never get expiresAt==null. But you can get a NumberFormatException. 
I would replace this else if-block with 
```
                } else if (expiresStr != null && !expiresStr.isEmpty()) {
                    String expiresStrVal = expiresStr.split("~")[0];
                    try {
                        expiration = Long.parseLong(expiresStrVal) * 1000;
                        LOGGER.info("Expires-based expiration {} ms", expiration);
                    } catch (NumberFormatException e) {
                        expiration = -1;
                        LOGGER.warn("Invalid Expires param in CDN url: {}", url);
                    }

                 } else {
```